R/unleash.R

```diff
@@ -100,3 +130,27 @@ handle_text <- function(text) {
     }
   }
 }
+
+# https://stackoverflow.com/questions/74292177/split-a-string-and-keep-delimiter
+split_sentences <- function(xml_node) {
+  sentences <- unlist(strsplit(
+    xml2::xml_text(xml_node),
+    "(?<=.)(?<=[\\?\\!\\.])",
+    perl = TRUE
+  ))
+  if (length(sentences) < 2) {
+    return(xml2::xml_text(xml_node))
+  }
+  # now we have to put back together the punctuation that was split
+  punctuations <- which(sentences %in% c(".", "!", "?"))
+  punctuations <- setdiff(punctuations, 1)
+
+  for (i in seq_len(length(punctuations))) {
```

Comment:
```suggestion
  for (i in seq_along(punctuations)) {
```

?

Or you could loop directly on punctuations I guess

```
for (punct in punctuations) {
```